drift/lib/src/runtime/types/mapping.dart

```diff
@@ -80,12 +83,19 @@ class SqlTypes {
 
   /// Maps the [dart] value into a SQL literal that can be embedded in SQL
   /// queries.
-  String mapToSqlLiteral(Object? dart) {
+  String mapToSqlLiteral(
+    Object? dart, [
+    SqlDialect dialect = SqlDialect.sqlite,
```

Comment:
I wonder if this can be refactored so that the `dialect` is a field in this class instead of a parameter? It seems easy to forget passing this parameter around somewhere, I rather have it be in some central place. `SqlTypes` is marked `@internal`, so we can change its API without calling it a breaking change.

Maybe we can remove the `types` field from `DriftDatabaseOptions` and instead turn it into a `SqlTypes createMapping(SqlDialect dialect)` method. Then the `GenerationContext` could store the options associated with the current dialect by calling that method? That way the dialect only needs to be passed in one place.

Reply:
Making this change would be breaking - as I highlighted in the PR description.

The reason for this is that making it a field on the class would require knowing the dialect when the `SqlTypes` instance is constructed, which happens to be in `DriftDatabaseOptions`. `DriftDatabaseOptions` has no knowledge of the dialect of the database it is being attached to, so we would have to also add a parameter to `DriftDatabaseOptions` - another breaking change.

Of course, such a parameter could be made optional, but then that would require postgres users to change their database options to add a parameter, which seems redundant since only the executor should care about the dialect...

Therefore, since the `SqlTypes` methods are mostly used internally and not in many places (only when translating sql -> dart or dart -> sql), I think it's best to just have it as a parameter for now.